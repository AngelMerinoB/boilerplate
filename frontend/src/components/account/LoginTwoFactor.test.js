import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import { LoginTwoFactor } from './LoginTwoFactor';
Enzyme.configure({ adapter: new Adapter() });

const noop = () => {};

describe('<LoginTwoFactor />', () => {
  test('should render <LoginTwoFactor /> without errors', () => {
    const wrapper = Enzyme.shallow(
      <LoginTwoFactor twoFactor={{}} loginTwoFactor={noop} location={{}} loggedIn={false} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
