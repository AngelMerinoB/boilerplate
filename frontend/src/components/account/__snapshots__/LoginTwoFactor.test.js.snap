// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginTwoFactor /> should render <LoginTwoFactor /> without errors 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LoginTwoFactor
    location={Object {}}
    loggedIn={false}
    loginTwoFactor={[Function]}
    twoFactor={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Form
        ErrorList={[Function]}
        disabled={false}
        liveValidate={false}
        noHtml5Validate={true}
        noValidate={false}
        onSubmit={[Function]}
        safeRenderCompletion={false}
        schema={
          Object {
            "properties": Object {
              "token": Object {
                "title": "Please enter the 6-digit authentication code or a recovery code",
                "type": "string",
              },
            },
            "required": Array [
              "token",
            ],
            "title": "Two Factor Authentication",
            "type": "object",
          }
        }
        uiSchema={
          Object {
            "token": Object {},
          }
        }
      >
        <fieldset>
          <button
            className="btn btn-primary btn-block btn-lg"
            type="submit"
          >
            Log In
          </button>
        </fieldset>
      </Form>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "ErrorList": [Function],
        "children": Array [
          undefined,
          false,
          false,
          <fieldset>
            <button
              className="btn btn-primary btn-block btn-lg"
              type="submit"
            >
              Log In
            </button>
          </fieldset>,
        ],
        "disabled": false,
        "liveValidate": false,
        "noHtml5Validate": true,
        "noValidate": false,
        "onSubmit": [Function],
        "safeRenderCompletion": false,
        "schema": Object {
          "properties": Object {
            "token": Object {
              "title": "Please enter the 6-digit authentication code or a recovery code",
              "type": "string",
            },
          },
          "required": Array [
            "token",
          ],
          "title": "Two Factor Authentication",
          "type": "object",
        },
        "uiSchema": Object {
          "token": Object {},
        },
      },
      "ref": null,
      "rendered": Array [
        undefined,
        false,
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <button
              className="btn btn-primary btn-block btn-lg"
              type="submit"
            >
              Log In
            </button>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Log In",
              "className": "btn btn-primary btn-block btn-lg",
              "type": "submit",
            },
            "ref": null,
            "rendered": "Log In",
            "type": "button",
          },
          "type": "fieldset",
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Form
          ErrorList={[Function]}
          disabled={false}
          liveValidate={false}
          noHtml5Validate={true}
          noValidate={false}
          onSubmit={[Function]}
          safeRenderCompletion={false}
          schema={
            Object {
              "properties": Object {
                "token": Object {
                  "title": "Please enter the 6-digit authentication code or a recovery code",
                  "type": "string",
                },
              },
              "required": Array [
                "token",
              ],
              "title": "Two Factor Authentication",
              "type": "object",
            }
          }
          uiSchema={
            Object {
              "token": Object {},
            }
          }
        >
          <fieldset>
            <button
              className="btn btn-primary btn-block btn-lg"
              type="submit"
            >
              Log In
            </button>
          </fieldset>
        </Form>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "ErrorList": [Function],
          "children": Array [
            undefined,
            false,
            false,
            <fieldset>
              <button
                className="btn btn-primary btn-block btn-lg"
                type="submit"
              >
                Log In
              </button>
            </fieldset>,
          ],
          "disabled": false,
          "liveValidate": false,
          "noHtml5Validate": true,
          "noValidate": false,
          "onSubmit": [Function],
          "safeRenderCompletion": false,
          "schema": Object {
            "properties": Object {
              "token": Object {
                "title": "Please enter the 6-digit authentication code or a recovery code",
                "type": "string",
              },
            },
            "required": Array [
              "token",
            ],
            "title": "Two Factor Authentication",
            "type": "object",
          },
          "uiSchema": Object {
            "token": Object {},
          },
        },
        "ref": null,
        "rendered": Array [
          undefined,
          false,
          false,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <button
                className="btn btn-primary btn-block btn-lg"
                type="submit"
              >
                Log In
              </button>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Log In",
                "className": "btn btn-primary btn-block btn-lg",
                "type": "submit",
              },
              "ref": null,
              "rendered": "Log In",
              "type": "button",
            },
            "type": "fieldset",
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
